src/runtime/set.cpp

```diff
@@ -600,6 +652,87 @@ Box* setGt(BoxedSet* self, BoxedSet* rhs) {
     return setIssuperset(self, rhs);
 }
 
+Box* setContains(BoxedSet* self, Box* v) {
+    RELEASE_ASSERT(PyAnySet_Check(self), "");
+
+    if (PySet_Check(v)) {
```

Comment:
Sorry to be a stickler, but can we just use CPython's logic here?  Your code seems very reasonable, but since it's different there are going to be weird corner cases where we will do something different from cpython (ex: subclassing from set and adding a `__hash__` function that throws an exception.  or the fact that CPython will create a temporary frozenset object as the standin.)